addons/api/addon/utils/flatten-nested-object.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+/**
+ * Flattens a nested object into a single-level object.
+ * @param obj
+ * @param result
+ * @param parentPath
+ * @returns {Object}
+ */
+
+export const flattenObject = (obj, result = {}, parentPath = '') => {
+  for (const key in obj) {
+    // Check if the value is an object and it is not null
+    if (typeof obj[key] === 'object' && obj[key] !== null) {
```

Comment:
Won't this also include arrays with the `typeof` check? Ember also has a typeOf [function](https://api.emberjs.com/ember/release/functions/@ember%2Futils/typeOf) to check for specific values

Reply:
thanks for the link, I did not know ember has its own typeof, I will use that here